server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java

```diff
@@ -317,32 +317,47 @@ public static boolean audited(final CountyDashboard the_cdb,
    * @param round The round to update.
    */
   private static void updateRound(final CountyDashboard cdb,
-                                 final Round round) {
+                                  final Round round) {
     for (final Long cvrID : new HashSet<>(round.auditSubsequence())) {
       final Map<String, AuditReason> auditReasons = new HashMap<>();
       final Set<AuditReason> discrepancies = new HashSet<>();
       final Set<AuditReason> disagreements = new HashSet<>();
 
-      // FIXME extract-fn: findOrCreate
       CVRAuditInfo cvrai = Persistence.getByID(cvrID, CVRAuditInfo.class);
       if (cvrai == null) {
         cvrai = new CVRAuditInfo(Persistence.getByID(cvrID, CastVoteRecord.class));
       }
-      // extract-fn
 
       if (cvrai.acvr() != null) {
         // do the thing
         // update the round statistics as necessary
-
         for (final ComparisonAudit ca : cdb.comparisonAudits()) {
-          auditReasons.put(ca.contestResult().getContestName(), ca.auditReason());
-          if (!discrepancies.contains(ca.auditReason()) &&
-              ca.computeDiscrepancy(cvrai.cvr(), cvrai.acvr()).isPresent()) {
-            discrepancies.add(ca.auditReason());
+          final String contestName = ca.contestResult().getContestName();
+          AuditReason auditReason = ca.auditReason();
+
+          if (ca.isCovering(cvrID) && auditReason.isTargeted()) {
+            // If this CVR is interesting to this audit, the discrepancy
+            // should be in the audited contests part of the dashboard.
+            LOGGER.debug(String.format("[updateRound: CVR %d is covered in a targeted audit."
+                                       + " contestName=%s, auditReason=%s]",
+                                       cvrID, contestName, auditReason));
+            auditReasons.put(contestName, auditReason);
+          } else {
+            // Otherwise, let's put it in the unaudited contest bucket.
+            auditReason = AuditReason.OPPORTUNISTIC_BENEFITS;
```

Comment:
did you mean to overwrite the auditReason variable here?

Reply:
Yes, that's how it falls into the desired bucket of un-audited contest discrepancies. It's an (ab)use of an implementation detail without changing quite a bit more.